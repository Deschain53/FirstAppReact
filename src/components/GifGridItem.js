import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( {id,title: nombre} ) => {
    //console.log(id,title,url);

    return (
        <div className = "card">
            
            <p> {nombre}  </p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    //url: PropTypes.string.isRequired
}


//<img src = {url}   alt = {title} />