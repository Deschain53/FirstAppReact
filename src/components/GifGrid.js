import React from 'react';  //,{useState, useEffect} 
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

   
    const {data:images, loading} = useFetchGifs( category );

    console.log(loading);


    return (  //Para regresar una secci√≥n con el texto de la categoria
        <>
        <h3 className = "animate__animated animate__fadeIn"> {category}  </h3>

        { loading && <p className = "animate__animated animate__flash "> Loading </p> } 

      <div className = "card-grid">
                { 
                    images.map( img  => (
                        <GifGridItem 
                        key = {img.id}
                        {...img}
                        />
                    ))
                }  
        </div> 
        </>
    )
    
}

  
/*GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}*/




    //useEffect( ()=>{   //Para ejecutar la instruccion solo cuando el componente se renderize por primera vez, fue reemplazado por el hook useFetchGifs
    //    getGifs(category)
    //    .then(setImages)//(imgs => setImages( imgs ))
    //},[ category ])

    //getGifs();





                /*images.map( img => (          //Esta es una forma de hacerlo
                    <li key = {img.id}> {img.title} </li>
                    ))*/
                /*images.map( ({id,title}) => (   //Haciendo uso de la desestructuracion
                        <li key = {id}> {title} </li>
                ))   */


            //{loading ? 'Cargando...' : 'Data cargada'}   //OPerador tenario