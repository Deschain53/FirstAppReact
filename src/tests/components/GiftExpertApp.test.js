import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

const title = 'Un titulo';
const url = 'https.fd';
const wrapper = shallow( <GifGridItem title = {title} url = {url} /> );

describe('Pruebas en <GiftExpertApp />',() => {



    test('debe mostrar <GifExpertApp/> correctamente',() => {     
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    
    test('debe tener la imagen igual al url y alt de los props ', () => {
        const img = wrapper.find('img');
        //console.log(img.htlml);

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe tener card', () => {
        const div = wrapper.find('div');
        //const className = div.prop('clasName');

        expect(div.prop('className')).toBe('card');
        //expect(className.includes('card')).toBe(true);   //  true para que lo incluya, false para que no lo incluya
    })
    
    

})