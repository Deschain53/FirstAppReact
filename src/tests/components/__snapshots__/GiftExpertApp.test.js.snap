// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp /> debe mostrar <GifExpertApp/> correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Un titulo"
    src="https.fd"
  />
  <p>
     
    Un titulo
      
  </p>
</div>
`;
